trustchain-http/src/attestation_utils.rs

```diff
@@ -0,0 +1,1178 @@
+                .unwrap()
+                .as_str()
+                .unwrap()
+                .to_string(),
+        ));
+        Ok(challenge)
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone, IsEmpty)]
+pub struct ContentCRInitiation {
+    pub requester_did: Option<String>,
+}
+
+impl ContentCRInitiation {
+    pub fn new() -> Self {
+        Self {
+            requester_did: None,
+        }
+    }
+
+    fn is_complete(&self) -> bool {
+        return self.requester_did.is_some();
+    }
+}
+
+impl ElementwiseSerializeDeserialize for ContentCRInitiation {
+    /// Deserialize each field of the struct from a file. Fields are optional. If no files are found, return None.
+    fn elementwise_deserialize(
+        mut self,
+        path: &PathBuf,
+    ) -> Result<Option<ContentCRInitiation>, TrustchainCRError> {
+        let requester_details_path = path.join("requester_did.json");
+        self.requester_did = match File::open(&requester_details_path) {
+            Ok(file) => {
+                let reader = std::io::BufReader::new(file);
+                let deserialized = serde_json::from_reader(reader)
+                    .map_err(|_| TrustchainCRError::FailedToDeserialize)?;
+                Some(deserialized)
+            }
+            Err(_) => None,
+        };
+
+        // if self.temp_p_key.is_none() && self.requester_did.is_none() {
+        //     return Ok(None);
+        // }
+        if self.requester_did.is_none() {
+            return Ok(None);
+        }
+
+        Ok(Some(self))
+    }
+}
+
+#[derive(Debug, Serialize, Deserialize, IsEmpty)]
+pub struct ContentCRChallenge {
+    pub content_nonce: Option<HashMap<String, Nonce>>,
+    pub content_challenge_signature: Option<String>,
+    pub content_response_signature: Option<String>,
+}
+
+impl ContentCRChallenge {
+    pub fn new() -> Self {
+        Self {
+            content_nonce: None,
+            content_challenge_signature: None,
+            content_response_signature: None,
+        }
+    }
+    /// Returns true if all fields required for the challenge have a non-null value.
+    fn challenge_complete(&self) -> bool {
+        return self.content_nonce.is_some() && self.content_challenge_signature.is_some();
+    }
+    /// Returns true if all fields required for the challenge-response have a non-null value.
+    fn is_complete(&self) -> bool {
+        return self.challenge_complete() && self.content_response_signature.is_some();
+    }
+}
+
+impl ElementwiseSerializeDeserialize for ContentCRChallenge {
+    /// Deserialize each field of the struct from a file. Fields are optional. If no files are found, return None.
+    fn elementwise_deserialize(
+        mut self,
+        path: &PathBuf,
+    ) -> Result<Option<ContentCRChallenge>, TrustchainCRError> {
+        // content nonce(s)
+        let mut full_path = path.join("content_nonce.json");
+        self.content_nonce = match File::open(&full_path) {
```

Comment:
Consider refactor `std::fs::read_to_string`

Reply:
Added a TODO in [1d9c1fe](https://github.com/alan-turing-institute/trustchain/pull/162/commits/1d9c1fe58411e88e159077269791b955dce29743)